### ISSUE TITLE
Type annotation for extra_validator

### ISSUE THREAD

**Issue by progval** [sissaschool/xmlschema#291]
Hi,

I cannot find how to type-annotate a function I use as `extra_validator`. Either my function itself fails type-checking (first example), or passing it to `schema.validate()` does (second example):

```py
from typing import Iterator, Optional
from xml.etree import ElementTree
import xmlschema

document = ElementTree.fromstring("<id>http://example.org</id>")
schema = xmlschema.XMLSchema11("""\
<?xml version="1.0" encoding="utf-8"?>
<xsd:schema xmlns:xsd="http://www.w3.org/2001/XMLSchema">
  <xsd:element name="id" type="xsd:anyURI" />
</xsd:schema>
""")

# Annotated with xmlschema.XMLSchemaBase:

def extra_validator1(
    element: ElementTree.Element,
    xsd_element: xmlschema.XMLSchemaBase,
) -> Optional[Iterator[xmlschema.XMLSchemaValidationError]]:
    xsd_element.type.name  # error: "XMLSchemaBase" has no attribute "type"; maybe "types"?
    return None

schema.validate(document, extra_validator=extra_validator1)

# Annotated with xmlschema.validators.elements.Xsd11Element:

def extra_validator2(
    element: ElementTree.Element,
    xsd_element: xmlschema.validators.elements.Xsd11Element,
) -> Optional[Iterator[xmlschema.XMLSchemaValidationError]]:
    xsd_element.type.name
    return None

schema.validate(document, extra_validator=extra_validator2)  # error: Argument "extra_validator" to "validate" of "XMLSchemaBase" has incompatible type "Callable[[Element, Xsd11Element], Optional[Iterator[XMLSchemaValidationError]]]"; expected "Optional[Callable[[Element, XMLSchemaBase], Optional[Iterator[XMLSchemaValidationError]]]]"
```

Either way, the code works at runtime.

Thanks for your help

**Comment by brunato** [comment]
Hi,

current type annotation is wrong, I'm sorry. It should be:

```python
Callable[[ElementType, XsdElement], Optional[Iterator[XMLSchemaValidationError]]] 
```

I will fix this for the next release, and advise you when it's available. Thank you for reporting this.

**Comment by progval** [comment]
Thanks!

**Comment by brunato** [comment]
This problem should be fixed with release v1.11.0
